pkg/activitypub/resthandler/activityhandler.go

```diff
@@ -273,7 +278,12 @@ func (h *Activities) getPage(objectIRI, id *url.URL, refType spi.ReferenceType,
 		items[i] = vocab.NewObjectProperty(vocab.WithActivity(activity))
 	}
 
-	id, prev, next, err := h.getIDPrevNextURL(id, it.TotalItems(), options)
+	totalItems, err := it.TotalItems()
```

Comment:
@bstasyszyn With CouchDB, it doesn't seem to be possible to get the total items from a query in the same call as the query itself (at least, not with the types of queries we need to do and the page options we need). This means that when `TotalItems` gets called, it's doing a separate query into CouchDB, so if the database was modified since the iterator was originally created, the TotalItems can change. Is this an issue?

Reply:
It's not really an issue if totalItems changes since it may also change when the client asks for the next page. We just need to make sure (as I think you did) that querying for any given page will give you the same result set. The 'super inefficient' part kinda bothers me. I don't know if we're going to run CouchDB or Postgres in production (I'm hoping with Postgres you can do this with one call). I think for now we should get it working and later decide how to improve performance.

Reply:
Just to clarify: the "super inefficient" part refers to how I could get totalItems in a single REST call (which isn't what I do, since it would involve pulling all items down in memory and counting)

Reply:
The method I use now is as efficient as possible (with CouchDB), but isn't fixed in time (i.e. calling TotalItems over and over again can give a different result if the database has changed in the meantime)